scipy/stats/_variation.py

```diff
@@ -193,11 +197,11 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
         else:
             del shp[axis]
         if len(shp) == 0:
-            result = np.nan
+            result = np.asarray(np.nan)
```

Comment:
While we're at it, consider `_get_nan` to preserve the dtype, e.g.
https://github.com/scipy/scipy/blob/2f6008b0a3aea3ce9ce7ca85a0860dab28b7a5fd/scipy/stats/_stats_py.py#L524-L526

Reply:
Sounds good, thanks for noticing that!